src/runtime/virtcontainers/clh.go

```diff
@@ -515,13 +507,29 @@ func (clh *cloudHypervisor) hotplugRemoveBlockDevice(drive *config.BlockDrive) e
 	return err
 }
 
+func (clh *cloudHypervisor) hotplugRemoveVfioDevice(device *config.VFIODev) error {
+	cl := clh.client()
+	ctx, cancel := context.WithTimeout(context.Background(), clhHotPlugAPITimeout*time.Second)
+	defer cancel()
+
+	_, err := cl.VmRemoveDevicePut(ctx, chclient.VmRemoveDevice{Id: device.ID})
+
+	if err != nil {
+		err = fmt.Errorf("failed to hotplug remove vfio device %+v %s", device, openAPIClientError(err))
```

Comment:
Nit: I'd consider adding a colon just before the original error:

```suggestion
		err = fmt.Errorf("failed to hotplug remove vfio device %+v: %s", device, openAPIClientError(err))
```